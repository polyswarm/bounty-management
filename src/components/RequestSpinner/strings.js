const strings = {
  token: 'Request in progress'
};
export default strings;