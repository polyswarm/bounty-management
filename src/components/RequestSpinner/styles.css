@keyframes Nectar-Token-Spin {
  from { transform: rotate(0deg); }
  to { transform: rotate(360deg); }
}

.Request-Spinner {
  animation: Nectar-Token-Spin infinite 20s linear;
}