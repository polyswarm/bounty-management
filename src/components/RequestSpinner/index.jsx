// Vendor imports
import React, { Component } from 'react';
import PropTypes from 'prop-types';
// Bounty Manager imports
// Component imports
import strings from './strings.js';

class RequestSpinner extends Component {
  render() {
    const { requests } = this.props;

    return (
      <span className='RequestSpinner'>
        {requests && requests.length > 0 && (
          <img className='RequestSpinner-Icon'
            src='../public/img/nct-coin.svg'
            alt={strings.token}/>
        )}
      </span>
    );
  }

}
RequestSpinner.proptypes = {
  requests: PropTypes.array
};
export default RequestSpinner;