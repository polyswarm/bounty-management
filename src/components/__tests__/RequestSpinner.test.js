import React from 'react';
import {shallow, render, mount} from 'enzyme';
import {shallowToJson, mountToJson} from 'enzyme-to-json';
import RequestSpinner from '../RequestSpinner';

it('Should render without error', () => {
  const wrapper = shallow(<RequestSpinner/>);
  expect(shallowToJson(wrapper)).toMatchSnapshot();
});

it('shows the icon when there are requests', () => {
  const request = [
    {title: 'demo', id: 'test'}
  ];
  const wrapper = mount(<RequestSpinner requests={request}/>);

  expect(wrapper.find('.RequestSpinner-Icon')).toHaveLength(1);
  expect(mountToJson(wrapper)).toMatchSnapshot();
});
