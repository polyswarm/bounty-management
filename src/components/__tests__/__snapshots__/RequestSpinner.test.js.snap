// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render without error 1`] = `<React.Fragment />`;

exports[`shows the icon when there are requests 1`] = `
<RequestSpinner
  requests={
    Array [
      Object {
        "id": "test",
        "title": "demo",
      },
    ]
  }
>
  <img
    className="Request-Spinner"
    src="img/nct-coin.svg"
  />
</RequestSpinner>
`;
